### PR TITLE
Block Styles: Remove the block margin in the style selector

### PR DESCRIPTION
**Pull request by scruffian**
## Description
This removes margins on the block preview when used in the styles selector.

This is being added here: https://github.com/WordPress/gutenberg/blob/17e5c2d870d84fb6de48dcd5bc3c38cd0c0fb0d0/packages/block-library/src/editor.scss#L56

but it doesn't make much sense for previews in the block style selector.

## How has this been tested?
In chrome

## Screenshots 
Before:
Screenshot 2020-01-31 at 15 29 40

After:
Screenshot 2020-01-31 at 15 38 29

## Types of changes
Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. .

### CODE REVIEW

File: packages/block-editor/src/components/block-list/style.scss [hunk 1]
```diff
@@ -19,6 +19,9 @@
 	}
 }
 
+.block-editor-block-styles .editor-styles-wrapper .block-editor-block-list__block {
```
Comment (ZebulanStanphill) [inline packages/block-editor/src/components/block-list/style.scss]: A comment explaining why this style is here would be helpful to others working on this code in the future.

Comment (youknowriad) [reply packages/block-editor/src/components/block-list/style.scss]: Should this be moved to the `block-editor-block-styles` style file instead?

Comment (youknowriad) [reply packages/block-editor/src/components/block-list/style.scss]: Also did we test with blocks that have "inner blocks"?

Comment (scruffian) [reply packages/block-editor/src/components/block-list/style.scss]: Happy to move this to `block-editor-block-styles` but I thought it was better to live near the specific code its targeting. 

Is there an example of a block with inner blocks which has a style preview?

Comment (youknowriad) [reply packages/block-editor/src/components/block-list/style.scss]: > Is there an example of a block with inner blocks which has a style preview?

Yes, the columns, group, media & text blocks.

It seems like there's something going wrong with the blocks you're trying to preview. Should we dig deeper to try to understand why it doesn't behave as expected?

Comment (scruffian) [reply packages/block-editor/src/components/block-list/style.scss]: I don't see style previews for these blocks in the side bar.

Comment (youknowriad) [reply packages/block-editor/src/components/block-list/style.scss]: Right! they only have an example shown in the inserter, to trigger the sidebar, you can just register a fake style variation.

**Comment by scruffian** [comment]
I added stye variations to the media and text block and tested with and without this patch:

before:
Screenshot 2020-02-12 at 14 38 36

after:
Screenshot 2020-02-12 at 14 37 01

The patch reduces the margins, but it looks ok to me.

**Comment by scruffian** [comment]
I also tested this for the columns block, which behaves similarly:

before:
Screenshot 2020-02-12 at 14 49 37

after:
Screenshot 2020-02-12 at 14 47 48

**Comment by pento** [comment]
This looks good to me, but I'd like design feedback on it. @jasmussen, do you have thoughts?

**Comment by jasmussen** [comment]
Yeah, this is kind of an unfortunate challenge of this. The preview is literally the block, identical to what it is in the post content. And right now the baseline of margin is just applied to every single block, affecting this preview as well. Which is kind of unfortunate.

The thing we want to continue to move towards, is a place where no such baseline margin exists at all, and is instead applied by each block individually. This requires a lighter DOM with fewer React fragments, which Ella is working on. And it's actually coming down the pike.

That's a long-winded way of saying: yes, I think it's fine to ship this PR as is (👍 👍) — but also that there's a very high probability that this particular CSS will at some point grow stale/die off and need to be changed. If you need to merge this, then it looks like a good improvement, but please keep an eye on Ella's doing, notably PRs like this one: https://github.com/WordPress/gutenberg/pull/19912

**Review by pento — APPROVED** [review]
🎉

**Pull request merged**